homeassistant/components/broadlink/config_flow.py

```diff
@@ -98,6 +98,9 @@ async def async_step_user(self, user_input=None):
 
             LOGGER.error("Failed to discover the device at %s: %s", host, err_msg)
 
+            if self.source == config_entries.SOURCE_IMPORT:
+                return self.async_abort(reason=errors["base"])
+
```

Comment:
Aborting here seems reasonable to me, so we avoid duplicate flows when things go wrong.